qcore/decorators.py

```diff
@@ -48,6 +48,8 @@ class DecoratorBinder(object):
     def __init__(self, decorator, instance=None):
         self.decorator = decorator
         self.instance = instance
+        if instance is not None:
+            self.__self__ = instance
```

Comment:
Setting `__self__` only for bound methods is consistent with how normal Python methods behave, since unbound methods don't have `__self__`. On the other hand, it's a bit weird for an attribute to only be present on the same type of object some of the time, so maybe we should just set `__self__` to None in the unbound case.

What do you think @besfahbod?

Reply:
Thanks. Yeah, I agree. Here's my reasoning:
- It does makes sense for PY to do so in the basic lang-item object types, keeping the API surface as small as possible.
- In this library, however, we're dealing with higher-level objects, specially ones that can work on multiple lang-item types. If we did have generics support over those types, we probably could be more specific. But since we don't, it's good to keep the API consistent regardless of the type of lang-item.

Reply:
Sounds good, done